### ISSUE TITLE
Rect is null when TextSpan with GestureDetector inside LayoutBuilder 

### ISSUE THREAD

**Issue by ricnaaru** [flutter/flutter#31949]
## Steps to Reproduce

```
class Ex extends StatefulWidget {
  @override
  _ExState createState() => _ExState();
}

class _ExState extends State<Ex> {

  var xxt = TapGestureRecognizer()..onTap = () {
    print("tap tap tap!");
  };
  @override
  Widget build(BuildContext context) {
    List<TextSpan> lts = [TextSpan(style: TextStyle(color: Color(0xff000000), fontSize: 14.0), text: "When Thor's evil brother, Loki (Tom Hiddleston), gains access to the unlimited power of the energy cube called the Tesseract, Nick Fury (Samuel L. Jackson), director of S.H.I.E.L.D., initiates a superhero recruitment effort to defeat the unprecedented threat to Earth. Joining Fury's \"dream team\" are Iron Man (Robert Downey Jr.), Captain America (Chris Evans), the Hulk (Mark Ruffalo), Thor (Chris Hemsworth), the Black Widow (Scarlett Johansson) and Hawkeye (Jeremy Renner)."),
    TextSpan(style: TextStyle(color: Color(0xff2196f3)), recognizer: xxt, text: "https://www.youtube.com/watch?v=hA6hldpSTF8"),];

    return Scaffold(body: RichText(
      text: TextSpan(children: lts),
      maxLines: 3,
    ),);
  }
}
```

## Logs

```
flutter: ══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
flutter: The following NoSuchMethodError was thrown during a scheduler callback:
flutter: The getter 'left' was called on null.
flutter: Receiver: null
flutter: Tried calling: left
flutter: 
flutter: When the exception was thrown, this was the stack:
flutter: #0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:50:5)
flutter: #1      RenderParagraph.assembleSemanticsNode.buildSemanticsConfig (package:flutter/src/rendering/paragraph.dart:495:14)
flutter: #2      RenderParagraph.assembleSemanticsNode (package:flutter/src/rendering/paragraph.dart:518:72)
flutter: #3      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3354:13)
flutter: #4      _SyncIterator.moveNext (dart:core/runtime/libcore_patch.dart:151:12)
flutter: #5      List.addAll (dart:core/runtime/libgrowable_array.dart:192:13)
flutter: #6      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3351:16)
flutter: #7      _SyncIterator.moveNext (dart:core/runtime/libcore_patch.dart:151:12)
flutter: #8      List.addAll (dart:core/runtime/libgrowable_array.dart:192:13)
flutter: #9      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3351:16)
flutter: #10     _SyncIterator.moveNext (dart:core/runtime/libcore_patch.dart:151:12)
flutter: #11     List.addAll (dart:core/runtime/libgrowable_array.dart:192:13)
flutter: #12     _RootSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3245:16)
flutter: #13     _SyncIterator.moveNext (dart:core/runtime/libcore_patch.dart:151:12)
flutter: #14     Iterable.single (dart:core/iterable.dart:552:13)
flutter: #15     RenderObject._updateSemantics (package:flutter/src/rendering/object.dart:2398:7)
flutter: #16     PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:951:16)
flutter: #17     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:285:19)
flutter: #18     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:677:13)
flutter: #19     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:219:5)
flutter: #20     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:990:15)
flutter: #21     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:930:9)
flutter: #22     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:842:5)
flutter: #23     _invoke (dart:ui/hooks.dart:154:13)
flutter: #24     _drawFrame (dart:ui/hooks.dart:143:3)
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
flutter: Another exception was thrown: NoSuchMethodError: The getter 'left' was called on null.
```
## Flutter Doctor

```
[✓] Flutter (Channel stable, v1.2.1, on Mac OS X 10.13.6 17G4015, locale en-ID)
    • Flutter version 1.2.1 at /Users/richardogvt/Desktop/SDK/flutter
    • Framework revision 8661d8aecd (3 months ago), 2019-02-14 19:19:53 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/richardogvt/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.6.1

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 34.0.2
    • Dart plugin version 183.5901
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] Connected device (1 available)
    • iPhone 6s • 57564569-552E-4702-B8B1-86B6AE710385 • ios • iOS 10.2 (simulator)

• No issues found!
```

**Comment by ricnaaru** [comment]
@jonahwilliams @dnfield here you go guys

**Comment by dnfield** [comment]
More discussion at #31423

We're somehow not getting a `Rect` from the selection.

**Comment by joprice** [comment]
Maybe this is saying the same thing, but I get this error when I have `TextSpan` with a recognizer, where the text overflows its container. Is there a workaround for this? Is seems like this prevents making large blocks of interactive text.

**Comment by dnfield** [comment]
I expect to have a fix for this shortly.  Unfortunately in investigating a fix for it I found some other bugs that I hve to fix first :)

**Comment by joprice** [comment]
I tried master and I no longer get a crash, but I do still see the following error. Is this unrelated?:

```flutter: ══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
flutter: The following assertion was thrown during a scheduler callback:
flutter: 'package:flutter/src/semantics/semantics.dart': Failed assertion: line 1439 pos 12:
flutter: '!owner._nodes.containsKey(id)': is not true.
flutter:
flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
flutter: more information in this error message to help you determine and fix the underlying cause.
flutter: In either case, please report this assertion by filing a bug on GitHub:
flutter:   https://github.com/flutter/flutter/issues/new?template=BUG.md
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #2      SemanticsNode.attach (package:flutter/src/semantics/semantics.dart:1439:12)
flutter: #3      AbstractNode.adoptChild (package:flutter/src/foundation/node.dart:132:13)
flutter: #4      SemanticsNode._replaceChildren (package:flutter/src/semantics/semantics.dart:1366:11)
flutter: #5      SemanticsNode.updateWith (package:flutter/src/semantics/semantics.dart:1767:5)
flutter: #6      RenderParagraph.assembleSemanticsNode (package:flutter/src/rendering/paragraph.dart:869:10)
flutter: #7      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3460:13)
flutter: #8      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:152:12)
flutter: #9      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:145:32)
flutter: #10     List.addAll (dart:core-patch/growable_array.dart:193:13)
flutter: #11     _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3452:16)
flutter: #12     _SyncIterator.moveNext (dart:core-patch/core_patch.dart:152:12)
flutter: #13     Iterable.single (dart:core/iterable.dart:552:13)
flutter: #14     RenderObject._updateSemantics (package:flutter/src/rendering/object.dart:2406:7)
flutter: #15     PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:958:16)
flutter: #16     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:350:19)
flutter: #17     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:701:13)
flutter: #18     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:285:5)
flutter: #19     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1016:15)
flutter: #20     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:958:9)
flutter: #21     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:874:5)
flutter: #25     _invoke (dart:ui/hooks.dart:236:10)
flutter: #26     _drawFrame (dart:ui/hooks.dart:194:3)
flutter: (elided 5 frames from class _AssertionError and package dart:async)```

**Comment by dnfield** [comment]
Do you have a reproduction for that?

It probably should go in a new issue.  Feel free to cc me on it.

**Comment by joprice** [comment]
I don't. But I fork my project and narrow it down.

**Comment by lawonga** [comment]
Getting this as well

**Comment by dnfield** [comment]
@lawonga it would be more helpful if you indicated what channel/commit of Flutter you were on - it's entirely possible that you're on a commit earlier than the fix for this.

If not, please provide a sample reproduction, preferably in a new issue.  Feel free to tag me in it.

**Comment by lawonga** [comment]
@dnfield you are correct - I'm on an older sdk version.

Seemed like it was to do with the `TextSpan` having a `TapGestureRecognizer` with the text being a blank string as well. I fixed it with a workaround.
